### PR TITLE
Integrate ESLint and Correct Lint Issues

### PR DESCRIPTION
**Pull request by benjinus**
Fix eslint code validation errors

### CODE REVIEW

**Comment by justinwilaby** [comment]
@benjinus -

Thank you for helping out! Would it be too much to ask for a full EsLint integration into this project along with the fixes you provided?

**Comment by benjinus** [comment]
@justinwilaby -

eslint intergration

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    parser: '@typescript-eslint/parser',
+    plugins: [
+        '@typescript-eslint'
+    ],
+    extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/recommended'
+    ],
+    rules: {
+        '@typescript-eslint/no-this-alias': 'warn',
```
Comment (justinwilaby) [inline .eslintrc.js:12]: Since bitwise operators are typical, let's turn off this rule.

**Comment by justinwilaby** [comment]
Thank you for the contribution! Just one comment and it's good to merge.

**Comment by benjinus** [comment]
@justinwilaby lint rules on bitwise operators and 'this' variable have been turn off

**Review by justinwilaby — APPROVED** [review]

**Pull request merged**